### ISSUE TITLE
Separation of plotting and analysis functions 

### ISSUE THREAD

**Issue by leloup314** [SiLab-Bonn/testbeam_analysis#71]
Like in the `generate_pixel_mask` and `cluster_hits` function of  `hit_analysis`, plotting is done separately  from the analysis, with an independent plotting function from `plot_utils`. I guess this should be the general approach of handling plotting since it is much more conveinient to have separate functions in case one needs to plot something externally. Maybe we can implement this in the future for all functions of the analysis.

**Comment by DavidLP** [comment]
Yes, this is the way to do it. Right now there might be still some analysis parts in the plotting functions, since the separation of analysis and plotting was introduced when the code was already rather feature complete. Fell free to change that.
